core/src/autogluon/core/metrics/classification_metrics.py

```diff
@@ -366,3 +368,70 @@ def quadratic_kappa(y_true, y_pred):
             labels = np.arange(y_pred.shape[1])
         y_pred = np.argmax(y_pred, axis=-1)
     return cohen_kappa_score(y_true, y_pred, labels=labels, weights='quadratic')
+
+
+def customized_binary_roc_auc_score(y_true, y_score, **kwargs) -> float:
+    """
+    Functionally identical to sklearn.metrics.roc_auc_score for binary classification.
+    Streamlined for binary classification to be faster by ~5x by avoiding validation checks of the inputs.
+    We can do this in AutoGluon because we guarantee the data is of proper form when entering this logic.
+
+    Parameters
+    ----------
+    y_true : array-like of int
+        Ground truth (correct) labels for n_samples samples. shape = (n_samples,)
+        Valid sample values are 1 and 0.
+    y_score : array-like of float
+        The prediction probabilities. shape = (n_samples,)
+    **kwargs :
+        Any additional arguments. If not empty, will fallback to sklearn's implementation
+
+    Returns
+    -------
+    roc_auc_score : float
+        The roc_auc_score in higher_is_better format.
+    """
+    if isinstance(y_true, pd.Series):
+        y_true = y_true.values
+    if isinstance(y_score, pd.Series):
+        y_score = y_score.values
+    if kwargs:
+        return sklearn.metrics.roc_auc_score(y_true, y_score, **kwargs)
+
+    desc_score_indices = np.argsort(y_score, kind="mergesort")[::-1]
+    y_score = y_score[desc_score_indices]
+    y_true = y_true[desc_score_indices]
+
+    # keep only indices that have different values to speed up future computation
+    distinct_value_indices = np.where(np.diff(y_score))[0]
+    threshold_idxs = np.r_[distinct_value_indices, y_true.size - 1]
```

Comment:
the doc says `array-like of int` there's no `.size` on int

Reply:
nit: please add quick comment on what's going on here. `np.r_` is not commonly used API and it requires maintainers to do additional exploration when reading this code.

Nice trick, BTW

Reply:
"the doc says array-like of int there's no .size on int"

Assumed to be numpy array

Reply:
Updating types to clarify